src/system/execute.py

```diff
@@ -6,11 +6,15 @@
 
 import logging
 import subprocess
+from os import PathLike
+from typing import Any, Tuple, Union
 
 
-def execute(command, dir, capture=True, raise_on_failure=True):
+def execute(command: str, dir: Union[str, PathLike[str]], capture: bool = True, raise_on_failure: bool = True) -> Tuple[int, Any, Any]:
```

Comment:
Union types without different behavior based on the type don't make much sense to me.  Lets convert this to a `str` and patch up any usages